src/main/java/soconnect/logic/parser/todo/TodoShowCommandParser.java

```diff
@@ -0,0 +1,86 @@
+package soconnect.logic.parser.todo;
+
+import static java.util.Objects.requireNonNull;
+import static soconnect.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static soconnect.logic.parser.ArgumentTokenizer.tokenizeToList;
+import static soconnect.logic.parser.CliSyntax.INDICATOR_ALL;
+import static soconnect.logic.parser.CliSyntax.INDICATOR_PRIORITY;
+import static soconnect.logic.parser.CliSyntax.INDICATOR_TAG;
+import static soconnect.logic.parser.CliSyntax.PREFIX_ALL;
+import static soconnect.logic.parser.CliSyntax.PREFIX_PRIORITY;
+import static soconnect.logic.parser.CliSyntax.PREFIX_TAG;
+import static soconnect.model.Model.PREDICATE_SHOW_ALL_TODOS;
+
+import java.util.List;
+
+import soconnect.logic.commands.todo.TodoShowCommand;
+import soconnect.logic.parser.ArgumentTokenizer;
+import soconnect.logic.parser.Parser;
+import soconnect.logic.parser.ParserUtil;
+import soconnect.logic.parser.Prefix;
+import soconnect.logic.parser.exceptions.ParseException;
+import soconnect.model.tag.Tag;
+import soconnect.model.todo.Priority;
+import soconnect.model.todo.TodoContainsPriorityPredicate;
+import soconnect.model.todo.TodoContainsTagPredicate;
+
+/**
+ * Parses input arguments and creates a new {@code TodoShowCommand} object.
+ */
+public class TodoShowCommandParser implements Parser<TodoShowCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the {@code TodoShowCommand}
+     * and returns a {@code TodoShowCommand} object for execution.
+     *
+     * @throws ParseException If the user input does not conform with the expected format.
+     */
+    public TodoShowCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, TodoShowCommand.MESSAGE_USAGE));
+        }
+
+        List<ArgumentTokenizer.PrefixArgument> argList = tokenizeToList(args, PREFIX_ALL, PREFIX_PRIORITY,
+            PREFIX_TAG);
```

Comment:
```suggestion

        List<PrefixArgument> argList = tokenizeToList(args, PREFIX_ALL, PREFIX_PRIORITY,
            PREFIX_TAG);
```

Not really an issue, but just change this to static import to standardize with the rest.